### ISSUE TITLE
Show Menu when holding to a day cell

### ISSUE THREAD

**Issue by tiepvuvan** [patchthecode/JTAppleCalendar#253]
Our project currently have a requirement that show menu ("Copy, Delete, Cut...") when user hold a day cell.
I've tried to modify your code base(implement UICollectionViewDelegate for showing menu) but it didn't work.

```swift
public func collectionView(_ collectionView: UICollectionView, shouldShowMenuForItemAt indexPath: IndexPath) -> Bool {
    return true
} 
    
public func collectionView(_ collectionView: UICollectionView, canPerformAction action: Selector, forItemAt indexPath: IndexPath, withSender sender: Any?) -> Bool {
    return true
}
public func collectionView(_ collectionView: UICollectionView, performAction action: Selector, forItemAt indexPath: IndexPath, withSender sender: Any?) {
}
```

Hope you will support this feature. 
Thank you

**Comment by patchthecode** [comment]
You do not have to modify the code to do this.
All you have to do is add a long press gesture to the calendar -> `UILongPressGestureRecognizer`

An example of adding gestures can be found [here](https://patchthecode.github.io/AddingGestures)
You can also find implementation of long press gesture in the continuous range selection code located at the bottom of this page [here](https://patchthecode.github.io/RangeSelection).

Basically, once long press is activated, simply present your view.

**Comment by patchthecode** [comment]
If you need more help on this, then let me know.

**Comment by tiepvuvan** [comment]
It would be much easier if we implement default Apple menu. We don't have to handle long press gesture, finding target rect, target cell....

**Comment by patchthecode** [comment]
Code for handling long press is very short

```swift
override func viewDidLoad() {
   super.viewDidLoad()

   let longPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(didStartLongPressing(gesture:)))
   longPressGesture.minimumPressDuration = 0.2    // User has the flexibility to set the time
   calendarView.addGestureRecognizer(longPressGesture)
}

func didStartLongPressing(gesture: UILongPressGestureRecognizer) {
   if let cellState = calendarView.cellStatus(at: gesture.location(in: gesture.view!)) {
         // Code for shouldShowMenuForDate
         // Code for canPerformActionForDate
         // Code for performActionForDate
   }
}
```

Now what you said is true. It would be easier for developers if we implemented menus into the library as you have suggested. But remember, this library is being used by very many people. We would not want to create specific cases for some developers. 

Currently the library is so flexible that any developer can create the showing of menus however they want it to be. I we implemented a default behavior, then I think we will be making the library less flexible, right?

**Comment by tiepvuvan** [comment]
I've tried your code to get cellState, but i need to get current cell, and it always return nil
```
if let cellState = calendarView.cellStatus(at: sender.location(in: sender.view!)) {
      print(cellState.cell()) // nil
}
```

**Comment by patchthecode** [comment]
Its always nill. That's true. It's a bug.
I just fixed it on `master branch`.
Can you test it now?

Put this in your `Podfile`.
```
pod 'JTAppleCalendar', :git => 'https://github.com/patchthecode/JTAppleCalendar.git'
```

and then do a `pod install`.

**Comment by tiepvuvan** [comment]
Thank you, it's worked.

**Comment by rajhcss** [comment]
I am still getting nil in the following function ```print(cellState.cell())```

**Comment by patchthecode** [comment]
@rajhcss provide a sample project?

**Comment by rajhcss** [comment]
Hi JT, thanks for your reply.  I probably found the issue.  I modified your code slightly and it seems to work.  Sorry I cannot provide the source code that I am working on.  Please find below the modified function in the UserInteractionFunctions.swift

```    /// Returns the cell status for a given point
    /// - Parameter: point of the cell you want to find
    /// - returns:
    ///     - CellState: The state of the found cell
    public func cellStatus(at point: CGPoint) -> CellState? {
        if let indexPath = indexPathForItem(at: point) {
            let cell = self.collectionView(self, cellForItemAt: indexPath) as? JTAppleCell
            //let cell = cellForItem(at: indexPath) as? JTAppleCell
            return cellStateFromIndexPath(indexPath, cell: cell)
        }
        return nil
    }
```

**Comment by patchthecode** [comment]
awesome.

**Comment by patchthecode** [comment]
@Dyanu create a new issue. I am lost on the context you are experiencing.
I'll respond there.
